packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,205 @@
+
+React is working on fixes to these issues, but in the meantime, our wrappers take care of setting properties and listening to events for you.
+
+The `@lit-labs/react` package provides two main exports:
+
+-   `createComponent()` creates a React component that _wraps_ an existing web component. The wrapper allows you to set props on the component and add event listeners to the component like you would any other React component.
+
+-   `useController()` lets you use a Lit reactive controller as a React hook.  
+
+## createComponent
+
+The `createComponent()` function makes a React component wrapper for a custom element class. The wrapper correctly passes React `props` to properties accepted by the custom element and listens for events dispatched by the custom element.
+
+### Usage
+
+Import `React`, a custom element class, and `createComponent`.
+
+```js
+import React from 'react';
+import {createComponent} from '@lit-labs/react';
+import {MyElement} from './my-element.js';
+
+export const MyElementComponent = createComponent({
+  tagName: 'my-element',
+  elementClass: MyElement,
+  react: React,
+  events: {
+    onActivate: 'activate',
+    onChange: 'change',
+  },
+});
+```
+
+After defining the React component, you can use it just as you would any other React component.
+
+```jsx
+<MyElementComponent
+  active={isActive}
+  onActivate={(e) => setIsActive(e.active)}
+  onChange={handleChange}
+/>
+```
+
+{% aside "positive" "no-header" %}
+
+See it in action in the [React playground examples](/playground/#sample=examples/react-basics).
+
+{% endaside %}
+
+#### Options
+
+`createComponent` takes an options object with the following properties:
+
+- `tagName`: The custom element's tag name.
+- `elementClass`: The custom element class.
+- `react`: The imported `React` object. This is used to create the wrapper component with the user supplied `React`. This can also be an import of `preact-compat`.
+- `events`: An object that maps an event handler prop to an event name fired by the custom element.
+
+#### Using slots
+
+Children of component created with `createComponent()` will render to the default slot of the custom element.
+
+```jsx
+<MyElementComponent>
+  <p>This will render in the default slot.</p>
+</MyElementComponent>
+```
+
+To render the child to a specific named slot, the standard `slot` attribute can be added.
+
+```jsx
+<MyElementComponent>
+  <p slot="foo">This will render in the slot named "foo".</p>
+</MyElementComponent>
+```
+
+Since React components cannot have a `slot` attribute, they will need to be wrapped with another container element. Giving it a `display: contents;` style will make this container disappear.
```

Comment:
```suggestion
Since React components are not themselves HTML elements, they usually cannot directly have a `slot` attribute. To render into a named slot, the component will need to be wrapped with a container element that has a `slot` attribute. If a wrapper element interferes with styling, like for grid and flexbox layouts, giving it a `display: contents;` style ([See MDN for details](https://developer.mozilla.org/en-US/docs/Web/CSS/display#box)) will remove the container from rendering, and only render its children.
```